### PR TITLE
Run RGemini tests through Python to assess compatibility

### PR DESCRIPTION
**Pull request by julio-smh**
Run RGemini tests through Python to assess compatibility. Addresses #32

### CODE REVIEW

**Review by loffleraSMH — COMMENTED** [review]
This is fantastic! Thanks so much @julio-smh for implementing the additional workflow. I reviewed the changes and ran some tests to make sure any failed tests are caught correctly. Everything works as expected.

The only suggestions I have are about documentation:

First, could we add a brief section about this in the README to let users know how they can use Rgemini in Python (and what the caveats are)? The documentation should include the following:
1) Some brief demo code illustrating how users would call an Rgemini function in Python using rpy2.
2) A note saying that this has only been tested for functions with unit tests (and point users to the `tests` directory to check which functions this applies to).
3) A brief disclaimer that even for functions where unit tests exist, users should be cautious when running Rgemini in Python since the unit tests may not cover all use cases. Thus, we recommend performing additional checks to make sure function outputs are correct.

Second, let's add a brief bullet point in `NEWS.md` mentioning that unit tests are now also run in Python.

Finally, for developers: Let's add a brief note in `CONTRIBUTING.md` (maybe to point 6. unit tests) mentioning that the CI/CD workflow also runs all unit tests in Python.

Also, would it maybe be helpful to add some code developers can use to run individual unit tests locally during development (e.g., for debugging purposes)? Something like this (or maybe there is a better way)? 
```
from rpy2.robjects.packages import importr
from rpy2 import robjects
from rpy2.robjects import r
pkg_path = "path/to/Rgemini"
devtools = importr("devtools")
r('devtools::load_all()')
r(f'setwd("{pkg_path}")')
r('source("R/cohort_creation.R")') # source updated function code from .R script
r('source("tests/testthat/test-cohort_creation.R")') # run test for updated function
```
This is probably too detailed for `CONTRIBUTING.md` but we could potentially add a topic in the discussion forum about this and then link to this in `CONTRIBUTING.md`?? Just a suggestion.

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# Rgemini `develop`
+
+* **Testing improvements:**
+  * Unit tests are now also run in Python via rpy2 to ensure cross-language compatibility
```
Comment (loffleraSMH) [inline NEWS.md:4]: Nice, thanks for adding this. I just pushed a super small [commit](https://github.com/GEMINI-Medicine/Rgemini/commit/d7ca45d7de681ea646abd67394fcc9d15ffe01d6) to add this to a new `develop` section at the top of NEWS.md. We typically accumulate a few changes under `develop` before we release a new version eventually. There are still some other improvements in progress (including hopefully new unit tests on a dummy DB, which will make this more robust), so I'll keep your changes on `develop` for now and release this with the next version (hopefully by the end of the summer).

File: README.md [hunk 2]
```diff
@@ -69,6 +69,46 @@ readm <- readmission(
...
+
+[Here](https://github.com/GEMINI-Medicine/Rgemini/discussions/200) is how to use Rgemini functions in R snippets in Python.
+
+### Important Limitations and Disclaimers
+
+**Testing Coverage**: Python compatibility has only been tested for functions that have unit tests in the R package. To check which functions have test coverage, review the test files in the [`tests/testthat/`](https://github.com/GEMINI-Medicine/Rgemini/tree/main/tests/testthat) directory.
+
+**Use with Caution**: Even for functions where unit tests exist, users should exercise caution when running Rgemini in Python. The unit tests may not cover all possible use cases or edge conditions. We strongly recommend performing additional validation checks to ensure function outputs are correct for your specific data and use case.
+
+**Data Type Considerations**: Be aware that data types may behave differently between R and Python. Pay special attention to how dates, factors, and missing values are handled when transferring data between the two environments.
```
Comment (loffleraSMH) [inline README.md:110]: Ah nice, that's a good one to add!

File: README.md [hunk 3]
```diff
@@ -69,6 +69,46 @@ readm <- readmission(
...
+base = importr('base')
+rgemini = importr('Rgemini')
+
+# Example: Using the lunique function
+r_vector = robjects.IntVector([1, 1, 2, 2, 2, 3])
+result = rgemini.lunique(r_vector)
+print(f"Number of unique values: {result[0]}")
+```
+
+[Here](https://github.com/GEMINI-Medicine/Rgemini/discussions/200) is how to use Rgemini functions in R snippets in Python.
```
Comment (loffleraSMH) [inline README.md:102]: Very nice, thanks so much for adding all of this. Great idea to have a short example here and then a more detailed example in a discussion post. This is super helpful!

**Review by loffleraSMH — APPROVED** [review]
Thanks for adding the thorough and clear documentation @julio-smh! This is great, I just made one super small additional change (see comment below). If that looks ok to you, feel free to go ahead and merge this into `develop` and close the issue.
